pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java

```diff
@@ -128,64 +122,88 @@ public Dictionary getDictionary() {
 
   @Override
   public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
-    if (_intResult == null) {
-      _intResult = new int[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+    int numDocs = projectionBlock.getNumDocs();
+    Object ref = _result;
```

Comment:
None whatsoever, but I also discovered that we create a lot more of these than I think is reasonable per query, and I think it may be wise to cache them externally. In case something changes externally, I want this method invocation to be consistent, even if the updates are racy.

Reply:
Good point. Currently we create these LiteralTransformFunctions per literal per segment, which isn't really necessary. We can create them only once, and cache them into the `QueryContext` so that they can be shared across different literals and segments. That can be done in a separate PR